pkg/controller/vsphere/reconciler.go

```diff
@@ -296,12 +297,21 @@ func (r *Reconciler) delete() error {
 		}
 	}
 
-	if _, err := vm.powerOffVM(); err != nil {
-		return err
+	powerState, err := vm.getPowerState()
+	if err != nil {
+		return fmt.Errorf("can not determine %v vm power state", r.machine.GetName())
```

Comment:
Is it worth putting the original `err` as trailing to the `fmt.Errorf(` here, instead of swallowing it?

Reply:
yep worth to do. Good catch, thanks.